### PR TITLE
[TT-1345] option to use predeployed contracts in OCR tests

### PR DESCRIPTION
**Pull request by Tofel**
This PR adds the ability to use existing LINK token and OCRv1 / OCRv2 contracts in OCR soak/smoke & load tests. It allows to specify number of OCR contracts to deploy as well as deciding whether existing contracts should be use as-is or whether they should be configured.

This functionality is crucial for testing OCR on chains like zkSync, where we cannot deploy contracts using our testing code, but we can do it using other tools. Since interaction with zkSync contracts is possible using geth wrappers we can execute test logic there.

Tests executed:
- [x] 1 OCRv1 contract used by smoke test @ Polygon Amoy 
- [x] 1 OCRv2contract used by smoke test @ Polygon Amoy
- [x] 1 OCRv1 contract used by smoke test @ zkSync Era 
- [x] 1 OCRv2 contract used by smoke test @ zkSync Era 
- [x] 1 OCRv1 contract  used by **soak** test @ zkSync Era 
- [x] 2 OCRv1 contract  used by **soak** test @ zkSync Era 

New contract number configuration format:
```
## OLD
# for OCRv1
[Soak.OCR.Soak]
number_of_contracts=2

## NEW
# for OCRv1
[OCR.Common]
number_of_contracts=2

## OLD
# for OCRv2
[Soak.OCR2.Soak]
number_of_contracts=2

## NEW
# for OCRv2
[OCR2.Common]
number_of_contracts=2
```
In nutshell: previously we could configure number of contracts only for Soak tests, now we can configure it for ALL tests, because it's no longer tightly coupled to test type.

### CODE REVIEW

**Review by chudilka1 — COMMENTED** [review]
If that is a new change/feature, and there is no info about this in README, it would be nice to update the later and explain how it works. Thanks.

File: integration-tests/actions/actions.go [hunk 1]
```diff
@@ -609,29 +610,48 @@ func TrackForwarder(
...
 }
 
-// DeployOCRv2Contracts deploys a number of OCRv2 contracts and configures them with defaults
-func DeployOCRv2Contracts(
+// SetupOCRv2Contracts deploys a number of OCRv2 contracts and configures them with defaults
+func SetupOCRv2Contracts(
 	l zerolog.Logger,
 	seth *seth.Client,
-	numberOfContracts int,
+	ocrContractsConfig ocr.OffChainAggregatorsConfig,
```
Comment (lukaszcl) [inline integration-tests/actions/actions.go:617]: @Tofel I assume no secrets in this config? If yes, use test secrets https://github.com/smartcontractkit/chainlink-testing-framework/blob/main/config/README.md#test-secrets

File: integration-tests/actions/actions.go [hunk 2]
```diff
@@ -842,28 +862,48 @@ func DeployOCRv1Contracts(
...
+			if (contractCount+1)%ContractDeploymentInterval == 0 { // For large amounts of contract deployments, space things out some
+				time.Sleep(2 * time.Second)
+			}
 		}
-		ocrInstances = append(ocrInstances, &ocrInstance)
-		if (contractCount+1)%ContractDeploymentInterval == 0 { // For large amounts of contract deployments, space things out some
-			time.Sleep(2 * time.Second)
+	} else {
+		// Load contract wrappers
+		for _, address := range ocrContractsConfig.OffChainAggregatorsContractsAddresses() {
```
Comment (lukaszcl) [inline integration-tests/actions/actions.go:897]: @Tofel, out of curiosity, why do we use an interface like OffChainAggregatorsConfig? Wouldn’t it be simpler to use config methods or fields directly?

**Review by lukaszcl — DISMISSED** [review]
@Tofel added few questions but overall lgtm

File: integration-tests/actions/actions.go (hunk 1 above)
Comment (Tofel) [reply integration-tests/actions/actions.go:617]: no test secrets there, it's just contract addresses

File: integration-tests/actions/actions.go (hunk 2 above)
Comment (Tofel) [reply integration-tests/actions/actions.go:897]: I have some logic underneath that returns empty array if contracts are disabled on some level. That way the caller doesn't have to know anything about that logic.

**Comment by polygon** [comment]
Please stop mentioning me in these tickets and spamming my notifications, thank you.

**Review by davidcauchi — DISMISSED** [review]

**Review by gheorghestrimtu — DISMISSED** [review]

**Review by davidcauchi — DISMISSED** [review]

**Review by gheorghestrimtu — APPROVED** [review]

**Review by lukaszcl — APPROVED** [review]

**Pull request merged**